pkg/arvo/app/group-store.hoon

```diff
@@ -30,50 +67,201 @@
+      ?:  ?=(%black kind.u.perm)
+        :-  [%open ~ who.u.perm]
+        ~
+      :_  who.u.perm
+      *invite:policy
+    ::
+    ++  migrate-group
+      |=  pax=path
+      =/  members
+        (~(got by groups.old) pax)
+      =^  =policy  members
+        (migrate-permissions pax members)
+      =/  rid=resource
+        (resource-from-old-path pax)
+      =/  =tags
+        (~(put ju *tags) %admin entity.rid)
+      [rid members tags policy %.n]
+    ::
+    ++  migrate-permissions
+      |=  [pax=path ships=(set ship)]
+      ^-  [policy (set ship)]
+      =/  perm
+        (scry-group-permissions pax)
+      ?~  perm
+        [*invite:policy ships]
+      ?>  ?=(%white kind.u.perm)
+      [[%invite ~] (~(uni in ships) who.u.perm)]
+    ::
+    ++  scry-unmanaged-groups
+      ^-  (set path)
+      .^  (set path)
+        %gx
+        (scot %p our.bowl)
+        %permission-store
+        (scot %da now.bowl)
+        /keys/noun
+      ==
+    ::
+    ++  scry-group-permissions
+      |=  pax=path
+      ^-  (unit permission:permission-store)
+      .^  (unit permission:permission-store)
+        %gx
+        (scot %p our.bowl)
+        %permission-store
+        (scot %da now.bowl)
+        ;:  weld
+          /permission
+          pax
+          /noun
+        ==
+      ==
+    --
   ::
   ++  on-poke
     |=  [=mark =vase]
     ^-  (quip card _this)
     ?>  (team:title our.bowl src.bowl)
     =^  cards  state
-      ?:  ?=(%group-action mark)
-        (poke-group-action:gc !<(group-action vase))
+      ?:  ?=(?(%group-update %group-action) mark)
+        (poke-group-update:gc !<(update:store vase))
       (on-poke:def mark vase)
     [cards this]
   ::
   ++  on-watch
     |=  =path
     ^-  (quip card _this)
     ?>  (team:title our.bowl src.bowl)
-    |^
-    =/  cards=(list card)
-      ?+    path  (on-watch:def path)
-          [%all ~]   (give %group-update !>([%initial groups]))
-          [%updates ~]  ~
-          [%keys ~]  (give %group-update !>([%keys ~(key by groups)]))
-          [%group *]
-        (give %group-update !>([%path (~(got by groups) t.path) t.path]))
-      ==
-    [cards this]
-    ::
-    ++  give
-      |=  =cage
-      ^-  (list card)
-      [%give %fact ~ cage]~
-    --
+    ?>  ?=([%groups ~] path)
+    :_  this
+    [%give %fact ~ %group-update !>([%initial groups])]~
   ::
   ++  on-leave  on-leave:def
   ::
   ++  on-peek
     |=  =path
     ^-  (unit (unit cage))
     ?+  path  (on-peek:def path)
-      [%x *]  ``noun+!>((~(get by groups) t.path))
+        [%y %groups ~]
+      =/  =arch
+        :-  ~
+        %-  malt
+        %+  turn
+          ~(tap by groups)
+        |=  [rid=resource *]
+        ^-  [@ta ~]
+        =/  group=^path
+          (en-path:resource rid)
+        [(spat group) ~]
+      ``noun+!>(arch)
+    ::
+        [%x %groups %ship @ @ ~]
+      =/  rid=(unit resource)
+        (de-path-soft:resource t.t.path)
+      ?~  rid   ~
+      ``noun+!>((peek-group u.rid))
```

Comment:
Maybe with the addition of #3001 it might be worth writing marks for these, instead of using `%noun` everywhere?